### PR TITLE
Fixes #6864 - clarify job vs nohup behavior

### LINKED ISSUE #6864: Jobs gets killed when parent process exits
# Documentation Issue

See #6854 for more context. Need to add notes about Jobs being bound to the PowerShell process where created. And how to get `nohup`-like behavior.

Add new example using `Invoke-Command -AsJob` to create detached processes.

## Context of the issue

- URL to the article (Docs or GitHub URL):
- Issue affects the following content (check all that apply):

**Conceptual content**
- [ ] Overview and Install
- [ ] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Sample scripts
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [x] Version 7.1 preview content
- [x] Version 7.0 content
- [x] Version 6 content
- [x] Version 5.1 content

- Is the issue specific to a platform (Y/N - Win/macOS/Linux):

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes [AB#1792028](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1792028) - Fixes #6864 - clarify job vs nohup behavior

This PR restructures the about_ articles covering PowerShell jobs.
- Added notes about the parent/child relationships between jobs and the session that creates them
- Warn that jobs are terminated with the parent session exits
- Clarify the differences between "remote" jobs and thread-based jobs (eg. perf and serialization)
- Add an example for running jobs in a detached session using Invoke-Command
- Add a nohup example (with context) to Start-Process

## PR Context

Select the area of the Table of Contents containing the documents being changed.

**Conceptual content**
- [ ] Overview and Install
- [ ] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Sample scripts
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [x] Version 7.1 preview content
- [x] Version 7.0 content
- [x] Version 6 content
- [x] Version 5.1 content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md [hunk 1]
```diff
@@ -15,34 +15,47 @@ expression in the background without interacting with the current session.
...
-if the job takes an extended time to complete. You can continue to work in the
-session without interruption while the job runs.
+PowerShell concurrently runs commands and scripts through jobs. There are three
+jobs types provided by PowerShell to support concurrency.
+
+- `RemoteJob` - Commands and scripts run on a remote session. For information,
+  see [about_Remote_Jobs](about_Remote_Jobs.md).
+- `BackgroundJob` - Commands and scripts run in a separate process on the local
+  machine.
+- `PSTaskJob` or `ThreadJob` - Commands and script runs in a separate thread
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:25]: Shouldn't that be "and scripts run"?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md [hunk 2]
```diff
@@ -15,34 +15,47 @@ expression in the background without interacting with the current session.
...
+  within the same process on the local machine. For more information, see
+  [about_Thread_Jobs](/powershell/module/ThreadJob/about_Thread_Jobs).
+
+Running scripts remotely, on a separate machine or in a separate process,
+provide great isolation. Any errors that occur in the remote job do not affect
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:30]: provides

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md [hunk 3]
```diff
@@ -15,34 +15,47 @@ expression in the background without interacting with the current session.
...
+other running jobs or the parent session that started the job. But the remoting
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:31]: "But the remoting..." --> "However, the remoting..."

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md [hunk 4]
```diff
@@ -15,34 +15,47 @@ expression in the background without interacting with the current session.
...
+layer adds overhead, including object serialization. All objects are serialized
+and deserialized as they are passed between the parent session and the remote
+(job) session. Serialization of large complex data objects can consume large
+amounts of compute and memory resources.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:35]: Would it make sense to also comment on the potential bandwidth consumption here?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md [hunk 5]
```diff
@@ -15,34 +15,47 @@ expression in the background without interacting with the current session.
...
+
+Thread based jobs are not as robust as Remote and Background jobs, because they
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:37]: If the job types are getting proper cased, it should be done everywhere.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md [hunk 6]
```diff
@@ -15,34 +15,47 @@ expression in the background without interacting with the current session.
...
+run in the same process on different threads. If one job has a critical error
+that crashes the process, then all other jobs in the process are terminated.
+
+However, thread-based jobs require less overhead. They don't use the remoting
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:41]: thread-based is hyphenated here, but not on line 37.  Should be consistent.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md [hunk 7]
```diff
@@ -15,34 +15,47 @@ expression in the background without interacting with the current session.
...
+layer or serialization. The result objects are returned as references to live
+objects in the current session. Without this overhead, thread-based jobs run
+faster and use fewer resources than the other job types.
+
+> [!IMPORTANT]
+> The parent session that created the job monitors the job status and collects
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:47]: I think "job monitors" looks like a single noun, but it's not intended as such.  I'd reword as "The parent session that creates the job also monitors the job status and ..."

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md [hunk 8]
```diff
@@ -67,116 +80,95 @@ To start a background job on the local computer, use the `Start-Job` cmdlet.
...
+session without interruption while the job runs.
+
 The `Start-Job` command returns an object that represents the job. The job
 object contains useful information about the job, but it does not contain the
 job results.
 
-Save the job object in a variable, and then use it with the other Job cmdlets
-to manage the background job. The following command starts a job object and
-saves the resulting job object in the `$job` variable.
+Save the job object in a variable, and then use it with the other **Job**
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:98]: Can we reword the first sentence here?  It feels like a directive, and not an explanation.

**Comment by doctordns** [comment]
The new example in Start-Process  is meant to show the nohup behaviour on linus, like this:
```powershell
Start-Process nohup 'pwsh -noprofile -"1..120 | % { Write-Host . -NoNewline; sleep 1 }"'
```

Can this example be improved to show the parameter names. What parameter is 'nohup' a value for? or is this a weird LInux only incantation?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md [hunk 9]
```diff
@@ -240,33 +229,26 @@ Get-Job
...
 
-You can write the partial results to a file and then append newer results as
-they arrive or you can wait and check the state of the job later.
-
-You can use the **Wait** parameter of the `Receive-Job` cmdlet, which does not
-return the command prompt until the job is complete and all results are
-available.
+You can also use the **Wait** parameter of the `Receive-Job` cmdlet. When use
+use this parameter, the cmdlet does not return the command prompt until the job
+is complete and all results are available.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:248]: I'd change complete to "completed" since the intent is that the job is past tense before **Wait** returns.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md [hunk 10]
```diff
@@ -1,371 +1,415 @@
...
+PowerShell concurrently runs commands and scripts through jobs. There are three
+jobs types provided by PowerShell to support concurrency.
 
-A background job is a command that runs asynchronously without interacting
-with the current session. The command prompt returns immediately, and you can
-continue to use the session while the job runs.
+- `RemoteJob` - Commands and scripts run in a remote session.
+- `BackgroundJob` - Commands and scripts run in a separate process on the local
+  machine. For more information, see [about_Jobs](about_Jobs.md).
+- `PSTaskJob` or `ThreadJob` - Commands and script runs in a separate thread
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:23]: "and scripts run"

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md [hunk 11]
```diff
@@ -1,371 +1,415 @@
...
+  within the same process on the local machine. For more information, see
+  [about_Thread_Jobs](/powershell/module/ThreadJob/about_Thread_Jobs).
 
-By default, background jobs run on the local computer. However, you can use
-several different procedures to run background jobs on remote computers.
+Running scripts remotely, on a separate machine or in a separate process,
+provide great isolation. Any errors that occur in the remote job do not affect
+other running jobs or the parent session that started the job. But the remoting
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:29]: "But the remoting..." --> "However, the remoting..."

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md [hunk 12]
```diff
@@ -1,371 +1,415 @@
...
+and deserialized as they are passed between the parent session and the remote
+(job) session. Serialization of large complex data objects can consume large
+amounts of compute and memory resources.
 
-This topic explains how to run a background job on a remote computer. For
-information about how to run background jobs on a local computer, see
-[about_Jobs](about_Jobs.md). For more information about background jobs, see
-[about_Job_Details](about_Job_Details.md).
+> [!IMPORTANT]
+> The parent session that created the job monitors the job status and collects
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:36]: I think "job monitors" looks like a single noun, but it's not intended as such. I'd reword as "The parent session that creates the job also monitors the job status and ..."

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md [hunk 13]
```diff
@@ -1,371 +1,415 @@
...
-To view the contents of the PsLog.txt file on the Server01 computer at any
-time, start another interactive session, or run a remote command. This type of
-command is best run in a PSSession (a persistent connection) in case you want
-to use several commands to investigate and manage the data in the PsLog.txt
-file. For more information about PSSessions, see about_PSSessions.
+1. To view the contents of the PsLog.txt file on the Server01 computer at any
+   time, start another interactive session, or run a remote command. This type
+   of command is best run in a PSSession (a persistent connection) in case you
+   want to use several commands to investigate and manage the data in the
+   PsLog.txt file. For more information about PSSessions, see about_PSSessions.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:156]: File name should be formatted.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md [hunk 14]
```diff
@@ -1,371 +1,415 @@
...
-The following command uses the New-PSSession cmdlet to create a PSSession that
-is connected to the Server01 computer. The command saves the PSSession in the
-\$s variable.
+From the perspective of the job, all operations are local. You are just running
+commands remotely to manage a local job on the remote computer.
+
+1. Use the `Invoke-Command` cmdlet to run a `Start-Job` command on a remote
+   computer.
+
+   This command requires a PSSession (a persistent connection). If you use the
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:261]: Shouldn't PSSession be bolded (here and following)?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md [hunk 15]
```diff
@@ -1,371 +1,415 @@
...
+   Id       Name    State      HasMoreData     Location   Command
+   --       ----    -----      -----------     --------   -------
+   2        Job2    Running    True            Localhost  Get-Eventlog system
+   ```
+
+   When you run a `Start-Job` command remotely, `Invoke-Command` returns the
+   same type of job object that `Start-Job` returns. You can save the job
+   object in a variable, or you can use a `Get-Job` command to get the job.
+
+   Note that the value of the Location property shows that the job ran on the
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:296]: Location should be formatted.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md [hunk 16]
```diff
@@ -1,371 +1,415 @@
...
+   local computer, known as "LocalHost", even though the job ran on the
+   Server01 computer. Because the job object is created on the Server01
+   computer and the job runs on the same computer, it is considered to be a
+   local background job.
+
+1. To manage a remote background job, use the Job cmdlets. Because the job
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:302]: Should "Job cmdlets" be "`Job-*` cmdlets"?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md [hunk 17]
```diff
@@ -1,371 +1,415 @@
...
+
+   To see if the job is complete, use an `Invoke-Command` command to run a
+   `Get-Job` command in the PSSession that is connected to the Server01
+   computer.
+
+   ```powershell
+   Invoke-Command -session $s -scriptblock {Get-Job}
+   ```
+
+   The command returns a job object. The State property of the job object shows
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:314]: State should be formatted.

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md [hunk 18]
```diff
@@ -18,36 +18,59 @@ the current session process.
...
-Start a thread job by using the `Start-ThreadJob` cmdlet. This cmdlet is
-available in the **ThreadJob** module that ships with PowerShell.
-`Start-ThreadJob` returns a single job object that encapsulates the running
-command or script, and can be used with all PowerShell job manipulating
-cmdlets.
+- `RemoteJob` - Commands and scripts run in a remote session. For information,
+  see [about_Remote_Jobs](about_Remote_Jobs.md).
+- `BackgroundJob` - Commands and scripts run in a separate process on the local
+  machine. For more information, see [about_Jobs](about_Jobs.md).
+- `PSTaskJob` or `ThreadJob` - Commands and script runs in a separate thread
```
Comment (DCtheGeek) [inline reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:28]: "and scripts run"

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md [hunk 19]
```diff
@@ -18,36 +18,59 @@ the current session process.
...
+  within the same process on the local machine.
 
-## The job cmdlets
+Thread based jobs are not as robust as Remote and Background jobs, because they
```
Comment (DCtheGeek) [inline reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:31]: "Thread-based" and casing on job types.

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md [hunk 20]
```diff
@@ -18,36 +18,59 @@ the current session process.
...
+run in the same process on different threads. If one job has a critical error
+that crashes the process, then all other jobs in the process are terminated.
+
+However, thread-based jobs require less overhead. They don't use the remoting
+layer or serialization. The result objects are returned as references to live
+objects in the current session. Without this overhead, thread-based jobs run
+faster and use fewer resources than the other job types.
+
+> [!IMPORTANT]
+> The parent session that created the job monitors the job status and collects
```
Comment (DCtheGeek) [inline reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:41]: I think "job monitors" looks like a single noun, but it's not intended as such. I'd reword as "The parent session that creates the job also monitors the job status and ..."

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md [hunk 21]
```diff
@@ -18,36 +18,59 @@ the current session process.
...
-|`Receive-Job`    |Gets the results of jobs.                              |
-|`Stop-Job`       |Stops a running job.                                   |
-|`Wait-Job`       |Suppresses the command prompt until one or all jobs are|
-|                 |complete.                                              |
-|`Remove-Job`     |Deletes a job.                                         |
+There are two ways to start thread-based jobs:
 
-## How to start a thread job on the local computer
+- `Start-ThreadJob` - from the ThreadJob module
+- `ForEach-Object -Parallel -AsJob` - the parallel features added in PowerShell
```
Comment (DCtheGeek) [inline reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:59]: Should feature be plural? I'd say it was a single feature, but it was added as a number of parameters on cmdlets.

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md [hunk 22]
```diff
@@ -18,36 +18,59 @@ the current session process.
...
+  7.0
 
-To start a thread job on the local computer, use the `Start-ThreadJob` cmdlet.
+Use the same **Job** cmdlets described in [about_Jobs](about_Jobs.md) to manage
```
Comment (DCtheGeek) [inline reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:62]: `Job-*` instead of **Job**?

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md [hunk 23]
```diff
@@ -18,36 +18,59 @@ the current session process.
...
+thread-based jobs.
 
-To write a `Start-ThreadJob` command, enclose the command or script the job runs
-in curly braces (`{ }`).
+### Using `Start-ThreadJob`
 
-The following command starts a thread job that runs a `Get-Process` command on
+The ThreadJob module first shipped with PowerShell 6. It can also be installed
```
Comment (DCtheGeek) [inline reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:67]: Should ThreadJob be formatted?

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md [hunk 24]
```diff
@@ -59,6 +82,29 @@ running job. The job object contains useful information about the job including
 its current running status. It collects the results of the job as the results
 are being generated.
 
+### Using `ForEach-Object -Parallel -AsJob`
+
+PowerShell 7.0 added a new parameter set to the `ForEach-Object` cmdlet. The
+new parameters allow you to run script blocks in parallel thread as PowerShell
```
Comment (DCtheGeek) [inline reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:88]: threads

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md [hunk 25]
```diff
@@ -197,11 +233,27 @@ Start-ThreadJob { param ([string] $message) Write-Output $message } -ArgumentLis
...
+
+1..1 | ForEach-Object -Parallel { Write-Output $using:msg } -AsJob | Wait-Job | Receive-Job
+```
+
 Since thread jobs run in the same process, any variable reference type passed
 into the job has to be treated carefully. If it is not a thread safe object,
 then it should never be assigned to, and method and properties should never be
 invoked on it.
 
+The following example passes a thread safe dotNet `ConcurrentDictionary` object
```
Comment (DCtheGeek) [inline reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:252]: .NET

File: reference/6/Microsoft.PowerShell.Management/Start-Process.md [hunk 26]
```diff
@@ -128,6 +128,25 @@ Start-Process -FilePath "$env:comspec" -ArgumentList "/c dir `"%systemdrive%\pro
 Start-Process -FilePath "$env:comspec" -ArgumentList "/c","dir","`"%systemdrive%\program files`""
 ```
 
+### Example 8: Create a detached process on Linux
+
+On Windows, `Start-Process` creates an independent process in a new console window that remains
```
Comment (DCtheGeek) [inline reference/6/Microsoft.PowerShell.Management/Start-Process.md:133]: I'm not sure I'd say "new console window".  -NoNewWindow can prevent the new window and since it's just a process, it may not be a console at all.

File: reference/6/Microsoft.PowerShell.Management/Start-Process.md [hunk 27]
```diff
@@ -128,6 +128,25 @@ Start-Process -FilePath "$env:comspec" -ArgumentList "/c dir `"%systemdrive%\pro
...
+running independently of the launching shell. On non-Windows platforms, the newly started process is
+attached to the shell that launched. If the launching shell is closed, the child process is
+terminated.
+
+To avoid terminating the child process on Unix-like platforms, you can combine `Start-Process` with
+`nohup`. The following example launches a background PowerShell instance that stays alive even after
```
Comment (DCtheGeek) [inline reference/6/Microsoft.PowerShell.Management/Start-Process.md:139]: I might add that nohup is short for "no hangup". And I think the description can be improved.  The example doesn't launch a background PowerShell instance, it launches nohup as a background instance that through parameters launches a PowerShell instance (which isn't a child of the original shell process).

**Review by DCtheGeek — CHANGES_REQUESTED** [review]

**Comment by DCtheGeek** [comment]
> The new example in Start-Process is meant to show the nohup behaviour on linus, like this:
> 
> ```powershell
> Start-Process nohup 'pwsh -noprofile -"1..120 | % { Write-Host . -NoNewline; sleep 1 }"'
> ```
> 
> Can this example be improved to show the parameter names. What parameter is 'nohup' a value for? or is this a weird LInux only incantation?

I made more or less the same suggestion.  I think the example is fine, but the description of it needs clarifying.  `nohup` isn't a PowerShell parameter, it's the Linux command that PowerShell is starting with `Start-Process`. `nohup` than calls the **pwsh** process with arguments to ensure it's not a terminated child process in the event the original instance of PowerShell gets terminated.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md (hunk 1 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:25]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md (hunk 2 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:30]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md (hunk 3 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:31]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md (hunk 4 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:35]: Added

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md (hunk 5 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:37]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md (hunk 6 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:41]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md (hunk 7 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:47]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md (hunk 8 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:98]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md (hunk 9 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:248]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md (hunk 10 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:23]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md (hunk 11 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:29]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md (hunk 12 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:36]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md (hunk 13 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:156]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md (hunk 14 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:261]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md (hunk 16 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:302]: stet

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md (hunk 15 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:296]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md (hunk 17 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:314]: Fixed

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md (hunk 19 above)
Comment (sdwheeler) [reply reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:31]: fixed

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md (hunk 20 above)
Comment (sdwheeler) [reply reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:41]: fixed

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md (hunk 22 above)
Comment (sdwheeler) [reply reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:62]: stet

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md (hunk 23 above)
Comment (sdwheeler) [reply reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:67]: Fixed

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md (hunk 24 above)
Comment (sdwheeler) [reply reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:88]: fixed

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md (hunk 25 above)
Comment (sdwheeler) [reply reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md:252]: fixed

File: reference/6/Microsoft.PowerShell.Management/Start-Process.md (hunk 26 above)
Comment (sdwheeler) [reply reference/6/Microsoft.PowerShell.Management/Start-Process.md:133]: Fixed

File: reference/6/Microsoft.PowerShell.Management/Start-Process.md (hunk 27 above)
Comment (sdwheeler) [reply reference/6/Microsoft.PowerShell.Management/Start-Process.md:139]: fixed

**Comment by doctordns** [comment]
@DCtheGeek - i agree the example is fine, but we really should, here, produce examples with full parameter names.

It was not intuitively obvious what nohup was/is (I have little LInux background).  

Can this example be further improved?

File: reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md (hunk 21 above)
Comment (sdwheeler) [reply reference/6/Microsoft.PowerShell.Core/About/about_Thread_Jobs.md]: Fixed

**Review by DCtheGeek — APPROVED** [review]

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md [hunk 28]
```diff
@@ -15,34 +15,49 @@ expression in the background without interacting with the current session.
...
+faster and use fewer resources than the other job types.
+
+> [!IMPORTANT]
+> The parent session that created the job also monitors the job status and
+> collects pipeline data. The job child process is terminated by the parent
+> process once the job reaches a finished state. If the parent session is
+> terminated, all running child jobs are terminated along with their child
+> processes.
+
+There are two ways work around this limitation:
```
Comment (PaulHigin) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:54]: I wouldn't call this a 'limitation'.  It is just a 'by design' behavior.  If a user wants different behavior, like 'fire and forget', then there is Invoke-Command session disconnect/reconnect, or Start-Process which really has nothing to do with jobs at all.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md [hunk 29]
```diff
@@ -1,371 +1,418 @@
...
-```
+   ```
+   Server01\C:>
+   ```
 
-While the job runs, you can use the interactive session to run other commands,
-including other background jobs. However, you must keep the interactive
-session open until the job is completed. If you end the session, the job is
-interrupted, and the results are lost.
+1. To start a background job in the session, use the `Start-Job` cmdlet. The
```
Comment (PaulHigin) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:101]: I wonder if we should remove this, since 'about_jobs' talks about background jobs and this is supposed to be about remote jobs.

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md (hunk 29 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Remote_Jobs.md:101]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md (hunk 28 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Jobs.md:54]: fixed

**Review by PaulHigin — APPROVED** [review]
